src/Eshopworld.Messaging/TopicAdapter.cs

```diff
@@ -24,6 +25,7 @@ internal sealed class TopicAdapter<T> : ServiceBusAdapter<T>
         internal TopicClient Sender;
         internal ISubscription AzureTopicSubscription;
         internal string SubscriptionName;
+        internal AsyncReaderWriterLock senderLock = new AsyncReaderWriterLock();

```

Comment:
What is the naming strategy? Other classes use `SenderLock` instead... Might be worth keeping them in sync, also for the other internal members in the class.